src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.dto.Transaction;
+import core.basesyntax.model.OperationType;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.service.ReportService;
+import core.basesyntax.strategy.AddOperationHandler;
+import core.basesyntax.strategy.BalanceOperationHandler;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.PurchaseOperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitServiceImplTest {
+    private static FruitService fruitService;
+    private static Map<OperationType, OperationHandler> testMap;
+
+    @BeforeClass
+    public static void beforeClass() {
+        testMap = new HashMap<>();
+        testMap.put(OperationType.BALANCE, new BalanceOperationHandler());
+        testMap.put(OperationType.PURCHASE, new PurchaseOperationHandler());
+        testMap.put(OperationType.RETURN, new AddOperationHandler());
+        testMap.put(OperationType.SUPPLY, new AddOperationHandler());
+        fruitService = new FruitServiceImpl(testMap);
+    }
+
+    @After
+    public void setUp() {
+        Storage.getFruits().clear();
+    }
+
+    @Test
+    public void callMethodOnEmptyListName_Ok() {
+        List<Transaction> testList = new ArrayList<>();
+        fruitService.applyOperations(testList);
```

Comment:
let's check here that the storage is still empty

Reply:
✔️